db/external_sst_file_basic_test.cc

```diff
@@ -351,6 +353,72 @@ TEST_F(ExternalSSTFileBasicTest, BasicWithFileChecksumCrc32c) {
   DestroyAndRecreateExternalSSTFilesDir();
 }
 
+TEST_F(ExternalSSTFileBasicTest, KeyValueOrdering) {
+  SyncPoint::GetInstance()->SetCallBack(
+      "SSTFileWriter::Add", [&](void* arg) {
+          Slice key = *static_cast<Slice*>(arg);
+          assert (key.size() > 0);
+          char* buf = const_cast<char*>(key.data());
+          buf[0] = 0x0; // corrupt the first byte
```

Comment:
Minor: might consider to do buf[0]++ so that it is more clearly to be corrupted.